src/main/java/it/pagopa/transactions/services/v2/TransactionsService.java

```diff
@@ -197,4 +207,95 @@ public NewTransactionResponseDto.ClientIdEnum convertClientId(
                         }
                 ).orElseThrow(() -> new InvalidRequestException("Null value as input origin"));
     }
+
+    @CircuitBreaker(name = "ecommerce-db")
+    @Retry(name = "getTransactionInfo")
+    public Mono<it.pagopa.generated.transactions.v2.server.model.TransactionInfoDto> getTransactionInfo(
+                                                                                                        String transactionId,
+                                                                                                        UUID xUserId
+    ) {
+        log.info("Get Transaction Invoked with id {} ", transactionId);
```

Comment:
Since transactionId and xUserId are not marked as nullable in this function,
maybe we should add a check for them to not be null before using them?


```   
@CircuitBreaker(name = "ecommerce-db")
@Retry(name = "getTransactionInfo")
public Mono<it.pagopa.generated.transactions.v2.server.model.TransactionInfoDto> getTransactionInfo(
    String transactionId,
    UUID xUserId
) {
    validateInputs(transactionId, xUserId); <------
    
    return getBaseTransactionView(transactionId, xUserId)
        .switchIfEmpty(Mono.error(new TransactionNotFoundException(transactionId)))
        .map(this::buildTransactionInfoDtoFromView)
        .doOnError(e -> log.error("Error retrieving transaction info for id {}: {}", transactionId, e.getMessage()));
}



private void validateInputs(String transactionId, UUID xUserId) {
    if (transactionId == null || transactionId.isEmpty()) {
        throw new IllegalArgumentException("Transaction ID cannot be null or empty");
    }
    if (xUserId == null) {
        throw new IllegalArgumentException("User ID cannot be null");
    }
}
`

Reply:
Input validation is already performed by the validation framework at controller level.

transactionId is set as required parameter in openapi, so a null value can never happen here

xUserId, instead, is an optional value, set only for transactions made by authenticated user (IO app). Checkout transaction are guest transactions so there is no xUserId to be set here and no filter to be made against transaction